kolibri/plugins/learn/assets/src/composables/__tests__/useSearch.spec.js

```diff
@@ -0,0 +1,460 @@
+import { get, set } from '@vueuse/core';
+import VueRouter from 'vue-router';
+import Vue from 'vue';
+import { ContentNodeResource } from 'kolibri.resources';
+import { coreStoreFactory } from 'kolibri.coreVue.vuex.store';
+import { AllCategories, NoCategories } from 'kolibri.coreVue.vuex.constants';
+import useSearch from '../useSearch';
+import { normalizeContentNode } from '../../modules/coreLearn/utils';
+
+Vue.use(VueRouter);
+
+const name = 'not important';
+
+function prep(query = {}) {
+  const store = coreStoreFactory({
```

Comment:
No action is needed. Just wanted to note that importing the whole store machinery can often bring quite a lot of complexity behind scenes and am not sure if it's worth it only to mock `state.route`, especially in a composable that is otherwise quite independent of the rest of the store. Was there any particular reason for bringing the whole core module into tests when compared to using an empty mock store or router and defining only values and methods that are needed?

Reply:
I did this because it's more representative of what is actually happening in production, and I didn't want any weird surprises to happen there compared to the tests.

Ideally, I would have just used the [VueRouter composable functions](https://next.router.vuejs.org/guide/advanced/composition-api.html) in here to access the route and the router, and not had the complexity inside or outside the tests, but that requires upgrading to VueRouter 4.x, and that seemed like a step too far for now!

Reply:
> I did this because it's more representative of what is actually happening in production, and I didn't want any weird surprises to happen there compared to the tests

Thank you for the explanation, makes sense. I usually find balancing these things quite tricky. It sounds that this approach has the advantage of being more representative as you say.

Reply:
No need to upgrade Vue Router for this :smile: